ras_party/controllers/respondent_controller.py

```diff
@@ -125,17 +125,63 @@ def update_respondent_mark_for_deletion(email: str, session):
 @with_db_session
 def delete_respondents_marked_for_deletion(session):
     """
-    Deletes all the existing respondents and there associated data which are marked for deletion
+    Deletes all the existing respondents and their associated data which are marked for deletion
+
+    NOTE: We don't delete pending_surveys records as these are subject to their own scheduled deletion
+    An IntegrityError exception will be logged if the respondent record cannot be deleted due
+    to existing pending_surveys records.
 
     :param session A db session
     """
     respondents = session.query(Respondent).filter(Respondent.mark_for_deletion == True)  # noqa
+    logger.info("Preparing to delete respondent records", number_to_delete=respondents.count())
+    failed_deletion_count = 0
     for respondent in respondents:
-        session.query(Enrolment).filter(Enrolment.respondent_id == respondent.id).delete()
-        session.query(BusinessRespondent).filter(BusinessRespondent.respondent_id == respondent.id).delete()
-        session.query(PendingEnrolment).filter(PendingEnrolment.respondent_id == respondent.id).delete()
-        session.query(Respondent).filter(Respondent.id == respondent.id).delete()
-        send_account_deletion_confirmation_email(respondent.email_address, respondent.first_name)
+        bound_logger = logger.bind(
+            email=obfuscate_email(respondent.email_address),
+            respondent_id=respondent.id,
+            party_uuid=respondent.party_uuid,
+        )
+        bound_logger.info("Attempting to delete respondent records")
+        try:
+            delete_respondent_records(session, respondent)
+            send_account_deletion_confirmation_email(respondent.email_address, respondent.first_name)
+            bound_logger.info("Respondent records deleted successfully")
+        except IntegrityError as e:
+            logger.error(
+                "A data constraint violation occurred trying to delete the respondent records",
+                respondent_id=respondent.id,
+                party_uuid=respondent.party_uuid,
+                error=str(e),
+            )
+            session.rollback()
+            failed_deletion_count += 1
```

Comment:
Rollback any failed deletions so we can move onto the next respondent being deleted, maintain a sanity count of the failures.